pkg/snapstore/gcs_snapstore.go

```diff
@@ -58,14 +59,23 @@ const (
 // NewGCSSnapStore create new GCSSnapStore from shared configuration with specified bucket.
 func NewGCSSnapStore(config *brtypes.SnapstoreConfig) (*GCSSnapStore, error) {
 	ctx := context.TODO()
-	var opts []option.ClientOption
+	var opts []option.ClientOption // no need to explicitly set store credentials here since the Google SDK picks it up from the standard environment variable
+
+	if _, ok := os.LookupEnv(envSourceStoreCredentials); !ok { // do not set endpoint override when copying backups between buckets, since the buckets may reside on different regions
+		endpoint := strings.TrimSpace(os.Getenv(envStorageAPIEndpoint))
+		if endpoint != "" {
+			opts = append(opts, option.WithEndpoint(endpoint))
+		}
+	}
```

Comment:
```suggestion
                 if endpoint, isSet := os.LookupEnv(envStorageAPIEndpoint); isSet {
			opts = append(opts, option.WithEndpoint(strings.TrimSpace(endpoint)))
		}
	}
```

Reply:
Does `isSet` guarantee that endpoint will not be empty or a bunch of whitespaces? Probably not. That's the reason I've explicitly checked for endpoint being empty after trimming whitespaces, so that we don't configure the google client with an invalid endpoint by accident.

Reply:
os.LookupEnv():
```
// LookupEnv retrieves the value of the environment variable named
// by the key. If the variable is present in the environment the
// value (which may be empty) is returned and the boolean is true.
// Otherwise the returned value will be empty and the boolean will
// be false.
```

The env var can be present but still be empty, and `isSet` will be returned as `true`.

Reply:
ok, I don't have strong opinion on this.